src/celeritas/io/ImportOpticalMaterial.hh

```diff
@@ -147,6 +147,28 @@ struct ImportOpticalProperty
     }
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Store optical photon wavelength shifting properties.
+ */
+struct ImportWavelengthShift
+{
+    double mean_num_photons;  //!< Mean number of re-emitted photons
+    double time_constant;  //!< Time delay between absorption and re-emission
+    ImportPhysicsVector absorption_length;  //!< Absorption length [MeV, len]
+    ImportPhysicsVector component;  //!< Emission spectrum [MeV, MeV]
+
+    //! Whether all data are assigned and valid
+    explicit operator bool() const
+    {
+        return mean_num_photons > 0 && time_constant > 0
+               && static_cast<bool>(absorption_length)
+               && static_cast<bool>(component)
+               && absorption_length.vector_type == ImportPhysicsVectorType::free
+               && absorption_length.vector_type == component.vector_type;
```

Comment:
```suggestion
               && component.vector_type == absorption_length.vector_type;
```
Not sure which is the better for readability.

Reply:
I like your suggestion.